src/components/sidebar/Sidebar/index.tsx

```diff
@@ -11,54 +10,49 @@ import SidebarFooter from '@/components/sidebar/SidebarFooter'
 
 import css from './styles.module.css'
 import { trackEvent, OVERVIEW_EVENTS } from '@/services/analytics'
-import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
-import OwnedSafes from '../OwnedSafes'
 
 const Sidebar = (): ReactElement => {
-  const router = useRouter()
   const [isDrawerOpen, setIsDrawerOpen] = useState<boolean>(false)
-  const isSafeRoute = !!router.query?.safe
 
-  const onDrawerToggle = () => {
-    trackEvent({ ...OVERVIEW_EVENTS.SIDEBAR, label: isDrawerOpen ? 'Close' : 'Open' })
-    setIsDrawerOpen((prev) => !prev)
-  }
+  const onDrawerToggle = useCallback(() => {
+    setIsDrawerOpen((isOpen) => {
+      trackEvent({ ...OVERVIEW_EVENTS.SIDEBAR, label: isOpen ? 'Close' : 'Open' })
+
+      return !isOpen
+    })
```

Comment:
Why was this change necessary?

Reply:
To not create a new callback reference on each render.